var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -597,6 +597,13 @@ def enable_or_disable(variant, keyword="USE", var=None, newer=False):
             # https://github.com/pytorch/pytorch/issues/60332
             # env.set("USE_SYSTEM_XNNPACK", "ON")
 
+    def setup_run_environment(self, env):
+        # The cmake_prefix_paths property from below doesn't end up in the run
+        # environment that one gets with spack load py-torch
+        env.prepend_path("CMAKE_PREFIX_PATH", self.cmake_prefix_paths[0])
+        # Similarly the libraries do not live at their "standard location"
+        env.prepend_path("LD_LIBRARY_PATH", self.spec["py-torch"].libs.directories[0])
```

Comment:
Note that for anything that gets RPATH'ed we add all of these directories by default. My guess would be something isn't getting the RPATHs at all (possibly bypassing the wrappers when building some object). Something like `find <prefix> -name *.so | xargs readelf -d | grep -l -v -i "<some-spack-prefix-that-would-be-rpathed>"` might help you find which

Reply:
I think @tmadlener is trying to build something outside of Spack that depends on py-torch

Reply:
Yes, exactly. The problem is not so much building things that depend on `py-torch` with spack, but rather that `py-torch` does not set the "expected" runtime environment properly. Here expected is roughly "the same as other packages do", i.e. `CMAKE_PREFIX_PATH` (and `LD_LIBRARY_PATH` if configured) are set to the correct places.